### PR TITLE
19 2 cancel invite

### PR DESCRIPTION
**Pull request by Artem672**

### CODE REVIEW

File: Provis.Core/Services/WorkspaceService.cs [hunk 1]
```diff
@@ -296,5 +296,34 @@ public async Task<List<WorkspaceMemberDTO>> GetWorkspaceMembersAsync(int workspa
 
             return workspaceMembers;
         }
+
+        public async Task CancelInviteAsync(int id, int workspaceId, string userId)
+        {
+            var invite = await _inviteUserRepository
+                .Query()
+                .FirstOrDefaultAsync(i => i.Id == id &&
```
Comment (YevheniiVovk) [inline Provis.Core/Services/WorkspaceService.cs]: Please compare execution plan for selected query with just Id and getByKeyASync method and your realization

File: Provis.Core/Services/WorkspaceService.cs [hunk 2]
```diff
@@ -296,5 +296,34 @@ public async Task<List<WorkspaceMemberDTO>> GetWorkspaceMembersAsync(int workspa
...
+                        i.IsConfirm == null);
+
+            _ = invite ?? throw new HttpException(System.Net.HttpStatusCode.NotFound,
+                "Invite with Id not found or it already answered");
+
+            var user = await _userWorkspaceRepository
+                .Query()
+                .FirstOrDefaultAsync(u => u.WorkspaceId == workspaceId && u.UserId == userId);
+
+            if (user.RoleId == (int)WorkSpaceRoles.OwnerId || invite.FromUserId == userId)
```
Comment (YevheniiVovk) [inline Provis.Core/Services/WorkspaceService.cs]: We should convert RoleId to enum

Comment (Artem672) [reply Provis.Core/Services/WorkspaceService.cs]: Fixed

File: Provis.Core/Services/WorkspaceService.cs (hunk 1 above)
Comment (Artem672) [reply Provis.Core/Services/WorkspaceService.cs]: I tried to compare and as a result, there was almost no difference between them but "GetByKey" is a better way to get value.

**Review by YevheniiVovk — APPROVED** [review]

**Review by AndriyKovalets — APPROVED** [review]

**Pull request merged**